app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js

```diff
@@ -85,7 +85,12 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
       if (!data.text && (element.querySelector('img') || element.querySelector('svg') || element.tagName === 'IMG' || element.closest('svg'))) {
         data.text = 'image'
       }
-      var url = data.url || this.findLink(event.target).getAttribute('href')
+      var url = data.url || this.getLinkHref(event.target)
```

Comment:
This is a good solution and well done for spotting it, but it feels slightly too long. It feels like `getLinkHref` and `findLink` should be combined, since they're inherently related and only called here, but I see the problems involved with that. A shorter (but possibly lazier) solution could be to wrap the original code in a try/catch:

```JavaScript
try {
  var url = data.url || this.findLink(event.target).getAttribute('href')
} catch (e) {
  // no href found, so abort
  return
}
```

Reply:
Thanks @andysellick, I've updated the code to reflect this - should be ready for another review :+1: